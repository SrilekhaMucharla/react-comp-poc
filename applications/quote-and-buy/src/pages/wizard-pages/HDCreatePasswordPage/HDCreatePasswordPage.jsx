/* eslint-disable jsx-a11y/click-events-have-key-events */
/* eslint-disable jsx-a11y/no-noninteractive-element-interactions */
/* eslint-disable import/no-extraneous-dependencies*/
import React, { useState, useEffect } from 'react';
import { connect } from 'react-redux';
import { Container, Row, Col } from 'react-bootstrap';
import _ from 'lodash';
import { HastingsAccountService } from 'hastings-capability-account';
import PropTypes from 'prop-types';
import {
    HDLabelRefactor,
    HDInfoCardRefactor
} from 'hastings-components';
import { AnalyticsHDTextInput as HDTextInput, AnalyticsHDButton as HDButton } from '../../../web-analytics';
import eyeOff from '../../../assets/images/wizard-images/hastings-icons/icons/Icons_Eye-off.svg';
import eyeOn from '../../../assets/images/wizard-images/hastings-icons/icons/Icons_Eye-on.svg';
import tipCirclePurple from '../../../assets/images/icons/tip_circle_purple.svg';
import * as messages from './HDCreatePasswordPage.messages';

const HDCreatePasswordPage = ({
    toggleContinueElement,
    submissionVM,
    customizeSubmissionVM,
    onSetNewPassSuccess,
    multiCarFlag,
}) => {
    const [textInputScore, setTextInputScore] = useState();
    const [pswrdStrengthMessage, setPswrdStrengthMessage] = useState();
    const [pswrdStrengthDescription, setPswrdStrengthDescription] = useState();
    const [isContBtnDisable, setIsContBtnDisable] = useState(true);
    const [pswrd, setPswrd] = useState();
    const [error, setError] = useState(null);
    const pswrdToken = _.get(submissionVM, 'value.passwordToken', null);
    const [pswrdTokenValue, setPswrdTokenValue] = useState(null);
    const [isPasswordReveal, setPasswordReveal] = useState(false);
    const [blankSpacesInPassword, setBlankSpacesInPassword] = useState(false);

    useEffect(() => {
        toggleContinueElement(false);
    }, [toggleContinueElement]);

    useEffect(() => {
        if (!!pswrdToken && Object.keys(pswrdToken).length > 0 && pswrdToken.token.length > 0) {
            setPswrdTokenValue(pswrdToken.token);
        }
    }, [pswrdToken]);

    const handleChange = (event) => {
        const zxcvbn = require('zxcvbn');
        const { value } = event.target;
        if (value && value.length > 0) {
            if (value !== _.trim(value)) {
                setBlankSpacesInPassword(true);
                setIsContBtnDisable(true);
                setPswrdStrengthMessage();
                setPswrdStrengthDescription();
            } else {
                const result = (zxcvbn(value));
                setTextInputScore(result.score);
                setBlankSpacesInPassword(false);
                switch (result.score) {
                    case 0:
                        setPswrdStrengthMessage(messages.pswrdEasy);
                        setPswrdStrengthDescription(`${messages.pswrdEasyMessageOne}\n${messages.pswrdEasyMessageTwo}`);
                        setIsContBtnDisable(true);
                        break;
                    case 1:
                        setPswrdStrengthMessage(messages.pswrdEasy);
                        setPswrdStrengthDescription(`${messages.pswrdEasyMessageOne}\n${messages.pswrdEasyMessageTwo}`);
                        setIsContBtnDisable(true);
                        break;
                    case 2:
                        setPswrdStrengthMessage(messages.pswrdOk);
                        setPswrdStrengthDescription(messages.pswrdOkMessage);
                        setIsContBtnDisable(false);
                        break;
                    case 3:
                        setPswrdStrengthMessage(messages.pswrdStrong);
                        setPswrdStrengthDescription(messages.pswrdStrongMessage);
                        setIsContBtnDisable(false);
                        break;
                    case 4:
                        setPswrdStrengthMessage(messages.pswrdStrong);
                        setPswrdStrengthDescription(messages.pswrdStrongMessage);
                        setIsContBtnDisable(false);
                        break;
                    default:
                        setPswrdStrengthMessage();
                        setPswrdStrengthDescription();
                        break;
                }
            }
        } else {
            setBlankSpacesInPassword(false);
            setTextInputScore();
            setPswrdStrengthMessage();
            setPswrdStrengthDescription();
        }
        setPswrd(value);
    };


    const handleContinue = () => {
        if (multiCarFlag) {
            const parentPolicy = submissionVM.value.quotes.filter((qd) => qd.isParentPolicy)[0];
            const { passwordToken } = parentPolicy;
            const { accountHolder } = parentPolicy.baseData;
            const { dateOfBirth } = accountHolder;
            const { emailAddress1 } = accountHolder;
            const { quoteData } = parentPolicy;
            const brand = quoteData.offeredQuotes[0].branchCode;
            const request = {
                Brand: brand,
                DateOfBirth: `${dateOfBirth.day}/${dateOfBirth.month + 1}/${dateOfBirth.year}`,
                EmailAddress: emailAddress1,
                PasswordToken: passwordToken.token,
                NewPassword: pswrd,
                SetVerified: false
            };
            HastingsAccountService.setNewPassword(request)
                .then(() => {
                    onSetNewPassSuccess();
                })
                .catch(() => {
                    setError(messages.errorMessage);
                });
        } else {
            const dateObject = _.get(submissionVM, 'value.lobData.privateCar.coverables.drivers[0].dateOfBirth');
            const pswrdObj = {
                Brand: _.get(customizeSubmissionVM, 'value.quote.branchCode'),
                DateOfBirth: `${dateObject.day}/${dateObject.month + 1}/${dateObject.year}`,
                EmailAddress: _.get(submissionVM, 'value.baseData.accountHolder.emailAddress1'),
                PasswordToken: pswrdTokenValue,
                NewPassword: pswrd,
                SetVerified: false
            };
            HastingsAccountService.setNewPassword(pswrdObj)
                .then(() => {
                    onSetNewPassSuccess();
                })
                .catch(() => {
                    setError(messages.errorMessage);
                });
        }
    };

    return (
        <>
            <Row>
                <Col xs={12} className="wizard-head create-password-head">
                    <Container className="create-password-common-container">
                        <HDLabelRefactor Tag="h1" text={messages.pswrdHeader} />
                        <p className="mb-0">{messages.pswrdHeaderMessage}</p>
                    </Container>
                </Col>
            </Row>
            <Container className="password-contents create-password-common-container create-password-container">
                {error && (
                    <Row>
                        <Col>
                            <div className="error create-password__error">
                                {error}
                            </div>
                        </Col>
                    </Row>
                )}
                <Row>
                    <Col xs={12} md={7} className="px-mobile-0 create-password__password-col">
                        <div className="elevated-box create-password__password-wrapper">
                            <div className="create-password__password-wrapper__container">
                                <HDLabelRefactor
                                    Tag="h5"
                                    text={messages.createPswrd}
                                    id="create-password-password-header"
                                    className="create-password__password__header" />
                                <div className={
                                    `create-password__description ${textInputScore ? `create-password__description--${textInputScore}` : 'd-none'}`}
                                >
                                    <HDLabelRefactor
                                        Tag="span"
                                        text={pswrdStrengthDescription}
                                        className="font-medium" />
                                </div>
                                <HDTextInput
                                    webAnalyticsEvent={{ event_action: messages.summary, event_value: messages.newPswrd }}
                                    className="create-password__input"
                                    id="create-password-password-input"
                                    name="pass"
                                    type={isPasswordReveal ? 'text' : 'password'}
                                    appendContent={<img src={isPasswordReveal ? eyeOff : eyeOn} alt="" onClick={() => setPasswordReveal(!isPasswordReveal)} />}
                                    onChange={(event) => handleChange(event)}
                                    value={pswrd} />
                                <div className={
                                    `create-password__strength-bar create-password__strength-bar--set${textInputScore}${!textInputScore ? ' d-none' : ''}`}
                                >
                                    <div className={`create-password__current-strength create-password__current-strength--${textInputScore}`}>
                                        {pswrdStrengthMessage}
                                    </div>
                                </div>
                                {
                                    blankSpacesInPassword && (
                                        <div className="invalid-field margin-top-lg margin-top-lg-mobile mb-0">
                                            <div className="message">
                                                {messages.blankSpaces}
                                            </div>
                                        </div>
                                    )
                                }

                                <HDInfoCardRefactor
                                    id="create-password-info-card"
                                    image={tipCirclePurple}
                                    paragraphs={[
                                        messages.infoHeaderMessage,
                                        (
                                            <ul className="create-password__info-card__list mb-2">
                                                <li><span>{messages.infoMessageOne}</span></li>
                                                <li><span>{messages.infoMessageTwo}</span></li>
                                                <li><span>{messages.infoMessageThree}</span></li>
                                                <li><span>{messages.infoMessageFour}</span></li>
                                            </ul>
                                        )
                                    ]}
                                    theme="light"
                                    size="thin"
                                    className="create-password__info-card" />

                                <HDButton
                                    webAnalyticsEvent={{ event_action: messages.continueRedirect }}
                                    id="create-password-continue-btn"
                                    className="theme-white margin-top-md create-password__continue-btn"
                                    label={messages.continueButtonText}
                                    disabled={isContBtnDisable}
                                    onClick={() => handleContinue()} />
                            </div>
                        </div>
                    </Col>
                    <Col xs={12} md={5} className="px-mobile-0 create-password__my-acc-col">
                        <div className="elevated-box create-password__my-acc-container">
                            <HDLabelRefactor
                                Tag="h2"
                                text={messages.myAccount}
                                id="create-password-my-acc-header"
                                className="mt-0 create-password__my-acc__header" />
                            <HDLabelRefactor
                                Tag="h5"
                                text={messages.tryTheApp}
                                id="create-password-my-acc-subheader"
                                className="create-password__my-acc__subheader" />
                            <HDLabelRefactor
                                Tag="h5"
                                text={messages.manageInsurance}
                                id="create-password-my-acc-subheader"
                                className="create-password__my-acc__subheader" />
                            <ul className="pad-inl-start-sm mb-0">
                                <li>{messages.policyChanges}</li>
                                <li>{messages.managePayments}</li>
                                <li>{messages.trackClaim}</li>
                                <li>{messages.downloadDocuments}</li>
                            </ul>
                        </div>
                    </Col>
                </Row>
            </Container>
        </>
    );
};

const mapStateToProps = (state) => ({
    multiCarFlag: state.wizardState.app.multiCarFlag,
    submissionVM: state.wizardState.data.submissionVM,
    customizeSubmissionVM: state.wizardState.data.customizeSubmissionVM,
});

HDCreatePasswordPage.propTypes = {
    multiCarFlag: PropTypes.bool,
    toggleContinueElement: PropTypes.func,
    submissionVM: PropTypes.shape({ lobData: PropTypes.object, value: PropTypes.object }),
    customizeSubmissionVM: PropTypes.shape({ value: PropTypes.object }),
    onSetNewPassSuccess: PropTypes.func
};

HDCreatePasswordPage.defaultProps = {
    multiCarFlag: false,
    toggleContinueElement: () => { },
    submissionVM: null,
    customizeSubmissionVM: null,
    onSetNewPassSuccess: () => { },
};

export default connect(mapStateToProps, null)(HDCreatePasswordPage);
