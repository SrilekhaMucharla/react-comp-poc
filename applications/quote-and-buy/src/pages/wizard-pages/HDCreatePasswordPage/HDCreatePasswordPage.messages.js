export const pswrdHeader = 'Your payment has been taken, just one last thing...';
export const pswrdHeaderMessage = "Please give us a password so we can set up your account. You'll then have 24/7 access to your policy and documents.";
export const createPswrd = 'Create password';
export const newPswrd = 'New password';
export const pswrdOk = 'OK';
export const pswrdOkMessage = "Great! You're good to go with that password.";
export const pswrdStrong = 'Strong';
export const pswrdStrongMessage = 'Good work! This password is really strong.';
export const pswrdEasy = 'Not strong enough';
export const pswrdEasyMessageOne = 'Whoops. That password is easy to guess.';
export const pswrdEasyMessageTwo = 'Try adding more characters or words.';
export const infoHeaderMessage = 'Your password should be easy to remember (but hard to guess).\nConsider using:';
export const infoMessageOne = 'uncommon words';
export const infoMessageTwo = 'unpredictable uPPercaSing';
export const infoMessageThree = 'Kreativ spellin';
export const infoMessageFour = 'less obvious numb8rs or symb#ls';
export const blankSpaces = 'Please don\'t start or end with a blank space';

export const myAccount = 'MyAccount';
export const tryTheApp = 'Don\'t forget to try the app and MyAccount!';
export const manageInsurance = 'It\'s the hassle-free way to manage your insurance. You can:';
export const policyChanges = 'Make policy changes';
export const managePayments = 'Manage your payments';
export const trackClaim = 'Report or track a claim';
export const downloadDocuments = 'View and download documents';
export const continueButtonText = 'Continue';
export const errorMessage = 'Sorry, something\'s gone wrong.';
export const pswrd = 'Password';
export const summary = 'Summary - Payment processing';
export const continueRedirect = 'Continue - Redirecting from: HastingsDirectCreatePassword';
