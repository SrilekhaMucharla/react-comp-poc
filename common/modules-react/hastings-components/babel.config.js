module.exports = require('./../../../common/build-tools/gw-build-config-babel');
