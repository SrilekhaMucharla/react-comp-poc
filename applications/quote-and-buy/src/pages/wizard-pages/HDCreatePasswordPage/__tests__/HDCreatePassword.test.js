import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import _ from 'lodash';

// GW
import ViewModelServiceFactory from 'gw-portals-viewmodel-js/ViewModelServiceFactory';

// HD
import { HDLabelRefactor } from 'hastings-components';
import HDCreatePasswordPage from '../HDCreatePasswordPage';
import productMetadata from '../../../../generated/metadata/product-metadata.json';
import submission from '../../../../routes/SubmissionVMInitial';
import * as messages from '../HDCreatePasswordPage.messages';
import defaultTranslator from '../../__helpers__/testHelper';
import withTranslator from '../../__helpers__/test/withTranslator';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
/**
 * A helper function to perform onChange event on a given component.
 * @param {*} component - component to preform onChange() on.
 * @param {*} path - path prop to component.
 * @param {*} actionName - name prop.
 * @param {*} actionValue - new value of component 'value' prop.
 * @param {*} prevVal - previous value prop.
 */
const onChangeAction = (component, path, actionName, actionValue, prevVal = 'no') => {
    component
        .invoke('onChange')({
            target: {
                name: actionName,
                value: actionValue,
                previousvalue: prevVal,
                // eslint-disable-next-line no-unused-vars
                setAttribute: (name, value) => { /* mock */ },
                getAttribute: (attr) => (attr === 'path' ? path : '')
            }
        });
};

describe('<HDCreatePasswordPage />', () => {
    let store;
    let wrapper;
    beforeEach(() => {
        // Init VM
        const viewModelService = ViewModelServiceFactory.getViewModelService(
            productMetadata, defaultTranslator
        );

        // Initialize mockstore with empty state
        const submissionVM = viewModelService.create(
            submission,
            'pc',
            'edgev10.capabilities.quote.submission.dto.QuoteDataDTO'
        );

        const driverPath = 'lobData.privateCar.coverables.drivers.children[0]';

        const licenceTypeFieldname = 'licenceType.aspects.availableValues';
        const licenceTypePath = `${driverPath}.${licenceTypeFieldname}`;

        // this is an workaround, submissionVM is too big to create SNAP
        const aspects = _.get(submissionVM, licenceTypePath);
        _.set(submission, licenceTypePath, aspects);

        // Set default values
        const initialState = {
            wizardState: {
                data: {
                    submissionVM: submission,
                    pageId: 'test-page'
                },
                app: {
                    step: 1,
                    prevStep: 0
                },
            }
        };
        store = mockStore(initialState);
        wrapper = mount(withTranslator(
            <Provider store={store}>
                <HDCreatePasswordPage />
            </Provider>
        ));
    });

    it('render component ', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('verify HDLabelRefactor', () => {
        expect(wrapper.find('HDLabelRefactor').exists()).toBe(true);
    });

    it('verify InputGroupText', () => {
        expect(wrapper.find('InputGroupText').exists()).toBe(true);
    });

    it('verify classname create-password__current-strength', () => {
        expect(wrapper.find('.create-password__current-strength').exists()).toBe(true);
    });

    it('verify classname create-password__info-card', () => {
        expect(wrapper.find('.create-password__info-card').exists()).toBe(true);
    });

    it('verify classname create-password__my-acc-container', () => {
        expect(wrapper.find('.create-password__my-acc-container').exists()).toBe(true);
    });

    it('verify classname create-password__password-col', () => {
        expect(wrapper.find('.create-password__password-col').exists()).toBe(true);
    });

    it('verify classname create-password__error is hidden when setNewPassword is success', () => {
        expect(wrapper.find('.create-password__error').exists()).toBe(false);
    });

    it('enter new password - empty string', async () => {
        const input = wrapper.find('[id="create-password-password-input"]').at(0);
        await act(async () => onChangeAction(
            input, input.prop('path'), input.prop('name'), ''
        ));
        await act(async () => wrapper.update());

        expect(wrapper.find(HDLabelRefactor).at(2).prop('text'))
            .toBeUndefined();
    });

    it('enter new password - all lower case', async () => {
        const input = wrapper.find('[id="create-password-password-input"]').at(0);
        await act(async () => onChangeAction(
            input, input.prop('path'), input.prop('name'), 'test'
        ));
        await act(async () => wrapper.update());

        expect(wrapper.find(HDLabelRefactor).at(2).prop('text'))
            .toBe(
                `${messages.pswrdEasyMessageOne}\n${messages.pswrdEasyMessageTwo}`
            );
    });

    it('enter new password - all lower case and numbers', async () => {
        const input = wrapper.find('[id="create-password-password-input"]').at(0);
        await act(async () => onChangeAction(
            input, input.prop('path'), input.prop('name'), 'testpass11'
        ));
        await act(async () => wrapper.update());

        expect(wrapper.find(HDLabelRefactor).at(2).prop('text'))
            .toBe(
                `${messages.pswrdEasyMessageOne}\n${messages.pswrdEasyMessageTwo}`
            );
    });

    it('enter new password - OK', async () => {
        const input = wrapper.find('[id="create-password-password-input"]').at(0);
        await act(async () => onChangeAction(
            input, input.prop('path'), input.prop('name'), 'TEstpass11'
        ));
        await act(async () => wrapper.update());

        expect(wrapper.find(HDLabelRefactor).at(2).prop('text'))
            .toBe(messages.pswrdOkMessage);
    });

    it('enter new password - Strong', async () => {
        const input = wrapper.find('[id="create-password-password-input"]').at(0);
        await act(async () => onChangeAction(
            input, input.prop('path'), input.prop('name'), 'TEs3paSS_11'
        ));
        await act(async () => wrapper.update());

        expect(wrapper.find(HDLabelRefactor).at(2).prop('text'))
            .toBe(messages.pswrdStrongMessage);
    });

    it('enter new password - Strong II', async () => {
        const input = wrapper.find('[id="create-password-password-input"]').at(0);
        await act(async () => onChangeAction(
            input, input.prop('path'), input.prop('name'), 'W!ac23Q_f'
        ));
        await act(async () => wrapper.update());

        expect(wrapper.find(HDLabelRefactor).at(2).prop('text'))
            .toBe(messages.pswrdStrongMessage);
    });
});
