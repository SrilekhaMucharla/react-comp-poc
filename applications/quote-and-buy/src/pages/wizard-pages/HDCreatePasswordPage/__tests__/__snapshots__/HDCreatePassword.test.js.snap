// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HDCreatePasswordPage /> render component  1`] = `
<Component>
  <Provider
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <Connect(HDCreatePasswordPage)>
      <HDCreatePasswordPage
        customizeSubmissionVM={null}
        dispatch={[Function]}
        multiCarFlag={false}
        onSetNewPassSuccess={[Function]}
        submissionVM={
          Object {
            "baseData": Object {
              "accountHolder": Object {
                "dateOfBirth": null,
              },
              "isExistingCustomer": false,
              "isPostalDocument": false,
              "jobType": "Submission",
              "marketingContacts": Object {
                "allowEmail": false,
                "allowPost": false,
                "allowSMS": false,
                "allowTelephone": false,
              },
              "periodStatus": "Draft",
              "policyAddress": Object {},
              "producerCode": "Default",
              "productCode": "PrivateCar_Ext",
              "productName": "Private Car",
              "termType": "Annual",
            },
            "lobData": Object {
              "privateCar": Object {
                "coverables": Object {
                  "addInterestTypeCategory": "PAVhcleAddlInterest",
                  "drivers": Array [
                    Object {
                      "claimsAndConvictions": Object {
                        "claimsDetailsCollection": Array [],
                        "convictionsCollection": Array [],
                      },
                      "licenceObtainedDate": null,
                      "person": Object {
                        "prefix": Object {},
                      },
                      "previousPoliciesInformation": Object {},
                      "residingInUKSince": null,
                      "tempID": "dd309e0e-8153-423c-aaa5-9930481060ad",
                    },
                  ],
                  "vehicleDrivers": Array [
                    Object {},
                  ],
                  "vehicles": Array [
                    Object {
                      "isCarModified": "",
                      "ncdProtection": Object {
                        "drivingExperience": Object {},
                      },
                      "tracker": "",
                      "vehicleModifications": Array [],
                      "vehicleWorth": "",
                    },
                  ],
                },
                "preQualQuestionSets": Array [],
              },
            },
          }
        }
        toggleContinueElement={[Function]}
      >
        <Row
          noGutters={false}
        >
          <div
            className="row"
          >
            <Col
              className="wizard-head create-password-head"
              xs={12}
            >
              <div
                className="wizard-head create-password-head col-12"
              >
                <Container
                  className="create-password-common-container"
                  fluid={false}
                >
                  <div
                    className="create-password-common-container container"
                  >
                    <HDLabelRefactor
                      Tag="h1"
                      adjustImagePosition={true}
                      className=""
                      text="Your payment has been taken, just one last thing..."
                    >
                      <h1
                        className=""
                      >
                        Your payment has been taken, just one last thing...
                      </h1>
                    </HDLabelRefactor>
                    <p
                      className="mb-0"
                    >
                      Please give us a password so we can set up your account. You'll then have 24/7 access to your policy and documents.
                    </p>
                  </div>
                </Container>
              </div>
            </Col>
          </div>
        </Row>
        <Container
          className="password-contents create-password-common-container create-password-container"
          fluid={false}
        >
          <div
            className="password-contents create-password-common-container create-password-container container"
          >
            <Row
              noGutters={false}
            >
              <div
                className="row"
              >
                <Col
                  className="px-mobile-0 create-password__password-col"
                  md={7}
                  xs={12}
                >
                  <div
                    className="px-mobile-0 create-password__password-col col-md-7 col-12"
                  >
                    <div
                      className="elevated-box create-password__password-wrapper"
                    >
                      <div
                        className="create-password__password-wrapper__container"
                      >
                        <HDLabelRefactor
                          Tag="h5"
                          adjustImagePosition={true}
                          className="create-password__password__header"
                          id="create-password-password-header"
                          text="Create password"
                        >
                          <h5
                            className="create-password__password__header"
                            id="create-password-password-header"
                          >
                            Create password
                          </h5>
                        </HDLabelRefactor>
                        <div
                          className="create-password__description d-none"
                        >
                          <HDLabelRefactor
                            Tag="span"
                            adjustImagePosition={true}
                            className="font-medium"
                          >
                            <span
                              className="font-medium"
                            />
                          </HDLabelRefactor>
                        </div>
                        <ForwardRef
                          appendContent={
                            <img
                              alt=""
                              onClick={[Function]}
                              src="Icons_Eye-on.svg"
                            />
                          }
                          className="create-password__input"
                          id="create-password-password-input"
                          name="pass"
                          onBeforeOpen={null}
                          onBlur={null}
                          onCancel={null}
                          onChange={[Function]}
                          onClick={null}
                          onConfirm={null}
                          onDelete={null}
                          onEdit={null}
                          onFocus={null}
                          onSelect={null}
                          type="password"
                          webAnalyticsEvent={
                            Object {
                              "event_action": "Summary - Payment processing",
                              "event_value": "New password",
                            }
                          }
                        >
                          <ForwardRef
                            allowLeadingZero={false}
                            appendContent={
                              <img
                                alt=""
                                onClick={[Function]}
                                src="Icons_Eye-on.svg"
                              />
                            }
                            appendLabel={null}
                            className="create-password__input"
                            customClassName=""
                            data={null}
                            disabled={false}
                            icon={null}
                            id="create-password-password-input"
                            inputMode={null}
                            isInvalidCustom={null}
                            mask={null}
                            maskChar="_"
                            maxLength={null}
                            name="pass"
                            onBlur={[Function]}
                            onChange={[Function]}
                            onFocus={[Function]}
                            onKeyPress={[Function]}
                            path="path.to.input"
                            placeholder={null}
                            preText={null}
                            reset={false}
                            size={null}
                            svgIcon={null}
                            thousandSeprator={false}
                            tickIcon={false}
                            type="password"
                            value=""
                          >
                            <InputGroup
                              className="create-password__input "
                              size={null}
                            >
                              <div
                                className="create-password__input  input-group"
                              >
                                <FormControl
                                  className="hd-text-input__form-control"
                                  data-testid="text-input"
                                  disabled={false}
                                  id="create-password-password-input"
                                  inputMode={null}
                                  maxLength={null}
                                  name="pass"
                                  onBlur={[Function]}
                                  onChange={[Function]}
                                  onFocus={[Function]}
                                  onKeyPress={[Function]}
                                  path="path.to.input"
                                  placeholder={null}
                                  type="password"
                                  value=""
                                >
                                  <input
                                    className="hd-text-input__form-control form-control"
                                    data-testid="text-input"
                                    disabled={false}
                                    id="create-password-password-input"
                                    inputMode={null}
                                    maxLength={null}
                                    name="pass"
                                    onBlur={[Function]}
                                    onChange={[Function]}
                                    onFocus={[Function]}
                                    onKeyPress={[Function]}
                                    path="path.to.input"
                                    placeholder={null}
                                    type="password"
                                    value=""
                                  />
                                </FormControl>
                                <InputGroupAppend
                                  className="hd-text-input__input-group__append"
                                >
                                  <div
                                    className="hd-text-input__input-group__append input-group-append"
                                  >
                                    <InputGroupText
                                      className="hd-text-input__input-group__append__text"
                                    >
                                      <span
                                        className="hd-text-input__input-group__append__text input-group-text"
                                      >
                                        <img
                                          alt=""
                                          onClick={[Function]}
                                          src="Icons_Eye-on.svg"
                                        />
                                      </span>
                                    </InputGroupText>
                                  </div>
                                </InputGroupAppend>
                                <i
                                  className="fa fa-exclamation exclam hd-text-input__exclam"
                                />
                              </div>
                            </InputGroup>
                          </ForwardRef>
                        </ForwardRef>
                        <div
                          className="create-password__strength-bar create-password__strength-bar--setundefined d-none"
                        >
                          <div
                            className="create-password__current-strength create-password__current-strength--undefined"
                          />
                        </div>
                        <HDInfoCardRefactor
                          className="create-password__info-card"
                          id="create-password-info-card"
                          image="tip_circle_purple.svg"
                          isRight={false}
                          notes={null}
                          paragraphs={
                            Array [
                              "Your password should be easy to remember (but hard to guess).
Consider using:",
                              <ul
                                className="create-password__info-card__list mb-2"
                              >
                                <li>
                                  <span>
                                    uncommon words
                                  </span>
                                </li>
                                <li>
                                  <span>
                                    unpredictable uPPercaSing
                                  </span>
                                </li>
                                <li>
                                  <span>
                                    Kreativ spellin
                                  </span>
                                </li>
                                <li>
                                  <span>
                                    less obvious numb8rs or symb#ls
                                  </span>
                                </li>
                              </ul>,
                            ]
                          }
                          size="thin"
                          theme="light"
                          title={null}
                        >
                          <div
                            className="info-card left info-card--light info-card--thin create-password__info-card"
                            id="create-password-info-card"
                          >
                            <img
                              alt="info-card"
                              className="info-card__icon"
                              src="tip_circle_purple.svg"
                            />
                            <p
                              className="info-card__paragraph paragraph_0"
                              key="0"
                            >
                              Your password should be easy to remember (but hard to guess).
Consider using:
                            </p>
                            <p
                              className="info-card__paragraph paragraph_1"
                              key="1"
                            >
                              <ul
                                className="create-password__info-card__list mb-2"
                              >
                                <li>
                                  <span>
                                    uncommon words
                                  </span>
                                </li>
                                <li>
                                  <span>
                                    unpredictable uPPercaSing
                                  </span>
                                </li>
                                <li>
                                  <span>
                                    Kreativ spellin
                                  </span>
                                </li>
                                <li>
                                  <span>
                                    less obvious numb8rs or symb#ls
                                  </span>
                                </li>
                              </ul>
                            </p>
                          </div>
                        </HDInfoCardRefactor>
                        <ForwardRef
                          className="theme-white margin-top-md create-password__continue-btn"
                          disabled={true}
                          id="create-password-continue-btn"
                          label="Continue"
                          onBeforeOpen={null}
                          onBlur={null}
                          onCancel={null}
                          onChange={null}
                          onClick={[Function]}
                          onConfirm={null}
                          onDelete={null}
                          onEdit={null}
                          onFocus={null}
                          onSelect={null}
                          webAnalyticsEvent={
                            Object {
                              "event_action": "Continue - Redirecting from: HastingsDirectCreatePassword",
                            }
                          }
                        >
                          <HDButtonRefactor
                            className="theme-white margin-top-md create-password__continue-btn"
                            disabled={true}
                            fullWidth={false}
                            id="create-password-continue-btn"
                            label="Continue"
                            onClick={[Function]}
                            onKeyPress={null}
                            size={null}
                            type="button"
                            variant="primary"
                          >
                            <Button
                              active={false}
                              block={false}
                              className="hd-btn hd-btn-primary theme-white margin-top-md create-password__continue-btn"
                              disabled={true}
                              id="create-password-continue-btn"
                              onClick={[Function]}
                              onKeyPress={null}
                              size={null}
                              type="button"
                              variant="primary"
                            >
                              <button
                                className="hd-btn hd-btn-primary theme-white margin-top-md create-password__continue-btn btn btn-primary"
                                disabled={true}
                                id="create-password-continue-btn"
                                onClick={[Function]}
                                onKeyPress={null}
                                type="button"
                              >
                                Continue
                              </button>
                            </Button>
                          </HDButtonRefactor>
                        </ForwardRef>
                      </div>
                    </div>
                  </div>
                </Col>
                <Col
                  className="px-mobile-0 create-password__my-acc-col"
                  md={5}
                  xs={12}
                >
                  <div
                    className="px-mobile-0 create-password__my-acc-col col-md-5 col-12"
                  >
                    <div
                      className="elevated-box create-password__my-acc-container"
                    >
                      <HDLabelRefactor
                        Tag="h2"
                        adjustImagePosition={true}
                        className="mt-0 create-password__my-acc__header"
                        id="create-password-my-acc-header"
                        text="MyAccount"
                      >
                        <h2
                          className="mt-0 create-password__my-acc__header"
                          id="create-password-my-acc-header"
                        >
                          MyAccount
                        </h2>
                      </HDLabelRefactor>
                      <HDLabelRefactor
                        Tag="h5"
                        adjustImagePosition={true}
                        className="create-password__my-acc__subheader"
                        id="create-password-my-acc-subheader"
                        text="Don't forget to try the app and MyAccount!"
                      >
                        <h5
                          className="create-password__my-acc__subheader"
                          id="create-password-my-acc-subheader"
                        >
                          Don't forget to try the app and MyAccount!
                        </h5>
                      </HDLabelRefactor>
                      <HDLabelRefactor
                        Tag="h5"
                        adjustImagePosition={true}
                        className="create-password__my-acc__subheader"
                        id="create-password-my-acc-subheader"
                        text="It's the hassle-free way to manage your insurance. You can:"
                      >
                        <h5
                          className="create-password__my-acc__subheader"
                          id="create-password-my-acc-subheader"
                        >
                          It's the hassle-free way to manage your insurance. You can:
                        </h5>
                      </HDLabelRefactor>
                      <ul
                        className="pad-inl-start-sm mb-0"
                      >
                        <li>
                          Make policy changes
                        </li>
                        <li>
                          Manage your payments
                        </li>
                        <li>
                          Report or track a claim
                        </li>
                        <li>
                          View and download documents
                        </li>
                      </ul>
                    </div>
                  </div>
                </Col>
              </div>
            </Row>
          </div>
        </Container>
      </HDCreatePasswordPage>
    </Connect(HDCreatePasswordPage)>
  </Provider>
</Component>
`;
